packages/cozy-procedures/src/Procedure.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { Provider, connect } from 'react-redux'
+import get from 'lodash/get'
+
+import store from './redux/store'
+import Context from './redux/context'
+import creditApplicationTemplate from './templates/creditApplicationTemplate'
+import { init as initPersonalData } from './redux/personalDataSlice'
+
+class Procedure extends React.Component {
+  componentDidMount() {
+    this.props.initPersonalData(
+      get(creditApplicationTemplate, 'personalData.schema.properties', {})
+    )
+  }
+
+  render() {
+    return this.props.children
+  }
+}
+
+const mapDispatchToProps = dispatch => ({
+  initPersonalData: template => dispatch(initPersonalData(template))
+})
+
+const ProcedureContainer = connect(
+  null,
+  mapDispatchToProps,
+  null,
+  { context: Context }
+)(Procedure)
+
+const ProcedureWithStore = props => (
+  <Provider store={store} context={Context}>
+    <ProcedureContainer {...props} />
```

Comment:
This isn't great: because this component is the root component for the Router, it gets rerendered whenever a navigation happens. I don't think it's a problem to re-render the `Provider`… but it certainly isn't usual.